### DISCUSSION TITLE
Should unhandled group messages kill the sending actor?

### DISCUSSION THREAD

**Discussion by alatdneg** [actor-framework/actor-framework#1208] [Q&A]
I've got actors joining a group and some don't handle all the messages that can be received from it.

When the receiver prints and drops, the sender then becomes unresponsive.

receiver
*** unexpected message [id: 80, name: PlayheadSelectionActor]: (caf::event_atom(), caf::change_atom())

Something tries to talk to the sender.
[2020-06-04 10:31:41.070] [xstudio] [warning] xstudio::ui::qml::PlaylistUI::update_media()::<lambda(const caf::error&)> system_error(request_receiver_down)
[2020-06-04 10:31:41.070] [xstudio] [warning] xstudio::ui::qml::PlaylistUI::update_playheads()::<lambda(const caf::error&)> system_error(request_receiver_down)
[2020-06-04 10:31:41.484] [xstudio] [warning] Move to failed system_error(request_receiver_down).
[2020-06-04 10:31:41.510] [xstudio] [warning] Move to failed system_error(request_receiver_down).

It looks like if any subscriber to the group fails to handle messages sent to the group the original sender is killed in someway, like it's getting some error sent back to is from the receiver.

Currently I can get around this by handling every possible message sent to the group, but it's bit of a pain.

**Comment by uentity** [comment]
You're right about the source of this issue -- when actor can't handle incoming message, standard `print_and_drop` handler will send back an error message. Sender receives that message and dies,

You have two options: 1) install custom unexpected messages handler for receiver that **will not** send back an error message; or 2) set custom error message for sender (that will drop unexpected message errors). In my opinion it's preferable to go with first option.
See my question here: https://groups.google.com/forum/#!topic/actor-framework/qQBoNwkRob4

**Comment by Neverlord** [comment]
Actors don't know whether they've received a message by direct send or via group subscription (yet?).

So this is pretty much "working as intended" right now, although I can see why it may not be most intuitive behavior. Pub/sub certainly makes it much easier to end up with subscribers that won't be able to process everything that's sent to the group. It's also easy to overwhelm a sender with responses.

CAF could hide the original sender when sending through a group. Currently, having the original sender makes it easy to implementing rendezvous protocols, but maybe there's a better way / more things to consider.
